packages/airnode-protocol/contracts/utils/SelfServeRrpBeaconServerWhitelister.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../rrp/requesters/RrpBeaconServer.sol";
+import "./interfaces/ISelfServeRrpBeaconServerWhitelister.sol";
+
+/// @title Contract that allows to whitlist readers on the RrpBeaconServer
+/// @dev The SelfServeRrpBeaconServerWhitelister contract has the WhitelistExpirationSetterRole
+/// and the IndefiniteWhitelisterRole of the RrpBeaconServer contract. The deployer of this contract
+/// can specify the beaconIds for which readers can whitelist themselves. The deployer (also the owner)
+/// can also add new beaconIds later for readers to self whitelist themselves.
+contract SelfServeRrpBeaconServerWhitelister is
+    Ownable,
+    ISelfServeRrpBeaconServerWhitelister
+{
+    address public rrpBeaconServer;
+
+    mapping(bytes32 => uint64) public beaconIdToExpirationTimestamp;
+    mapping(bytes32 => bool) public beaconIdToIndefiniteWhitelistStatus;
+
+    /// @param rrpBeaconServerAddress The RrpBeaconServer contract to whitelist readers
+    /// @param beaconIdsExpirationTimestamp Array of beaconIds that have an associated expiration timestamp
+    /// @param expirationTimestamps Array of expiration timestamps that correspond to the beaconIds
+    /// @param beaconIdsIndefiniteWhitelistStatus Array of beaconIds that have an associated indefinite whitelist status
+    /// @param indefiniteWhitelistStatuses Array of indefinite whitelist status that correspond to the beaconIds
+    constructor(
+        address rrpBeaconServerAddress,
+        bytes32[] memory beaconIdsExpirationTimestamp,
+        uint64[] memory expirationTimestamps,
+        bytes32[] memory beaconIdsIndefiniteWhitelistStatus,
+        bool[] memory indefiniteWhitelistStatuses
```

Comment:
This would always be all `true`s so it's weird. Do this instead:
```
        address rrpBeaconServerAddress,
        bytes32[] memory beaconIds,
        uint64[] memory expirationTimestamps,
        bool[] memory indefiniteWhitelistStatuses
```

Reply:
Also validate that the argument lengths are equal

Reply:
I see what you mean, more like this then:
```    
bytes32[] memory beaconIdsExpirationTimestamp,
uint64[] memory expirationTimestamps,
bytes32[] memory beaconIdsIndefiniteWhitelistStatus
```

and not bool

Reply:
I mean you don't even need two lists of beacons, it could go like
```
[beaconId1, beaconId2, beaconId3]
[12345678, 0, 987654312]
[false, true, false]
```
Also `beaconIdsExpirationTimestamp` and `beaconIdsIndefiniteWhitelistStatus` are bad variable names. It should be something like `temporarilyWhitelistableBeaconIds` / `indefinitelyWhitelistableBeaconIds` (if you're going to go with https://github.com/api3dao/airnode/pull/789#discussion_r792637995)